src/Nethermind/Nethermind.JsonRpc/Modules/BlockFinderExtensions.cs

```diff
@@ -45,6 +45,11 @@ public static SearchResult<BlockHeader> SearchForHeader(this IBlockFinder blockF
 
         public static SearchResult<Block> SearchForBlock(this IBlockFinder blockFinder, BlockParameter? blockParameter, bool allowNulls = false)
         {
+            if (blockParameter is not null && (blockParameter.BlockHash < blockFinder.EarliestHash || blockParameter.BlockHash < blockFinder.EarliestHash))
```

Comment:
Same condition twice?
How do you want to compare hashes?

Reply:
🤦 forgot about the fact that there were hashes. Updated the code to fetch the block number instead